src/extension/src/azure/azure-auth/azureAuth.ts

```diff
@@ -66,15 +85,20 @@ export abstract class AzureAuth {
     }
   }
   public static async logout(): Promise<boolean> {
-    this.initialize();
-    if (this.api.status === CONSTANTS.AZURE_LOGIN_STATUS.LOGGED_IN) {
-      await commands.executeCommand("azure-account.logout");
-      // Make sure it did not return from timeout
+    const userConfirmation = await AzureAuth.promptUsersToLogout();
+    if (userConfirmation.payload.signOut) {
+      this.initialize();
       if (this.api.status === CONSTANTS.AZURE_LOGIN_STATUS.LOGGED_IN) {
-        throw new AuthorizationError(CONSTANTS.ERRORS.LOGOUT_FAILED);
+        await vscode.commands.executeCommand("azure-account.logout");
+        // Make sure it did not return from timeout
+        if (this.api.status === CONSTANTS.AZURE_LOGIN_STATUS.LOGGED_IN) {
+          throw new AuthorizationError(CONSTANTS.ERRORS.LOGOUT_FAILED);
+        }
       }
+      return true;
+    } else {
+      return false;
```

Comment:
Can we reduce nesting by starting with `if (!userConfirmation.payload.signOut) {return false}`?

Reply:
Yup, makes sense :)